orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/monitoreddeploy/MonitoredDeployBaseTask.java

```diff
@@ -54,22 +66,66 @@ public long getBackoffPeriod() {
 
   @Override
   public long getTimeout() {
-    // TODO(mvulfson): Use DeploymentMonitorDefinition
-    return TimeUnit.MINUTES.toMillis(30);
+    // NOTE: This is not used since we override getDynamicTimeout
+    return 0;
+  }
+
+  @Override
+  public long getDynamicTimeout(Stage stage) {
+    ensureMonitorDefinition(stage);
+
+    final Duration defaultTimeout = Duration.ofMinutes(60);
+    long timeout;
+
+    try {
+      timeout = TimeUnit.MINUTES.toMillis(monitorDefinition.getMaxAnalysisMinutes());
+    } catch (Exception e) {
+      log.error(
+          "Failed to compute timeout for {}, returning {} min",
+          getClass().getSimpleName(),
+          defaultTimeout.toMinutes(),
+          e);
+
+      timeout = defaultTimeout.toMillis();
+    }
+
+    return timeout;
+  }
+
+  @Override
+  public @Nullable TaskResult onTimeout(@Nonnull Stage stage) {
+    ensureMonitorDefinition(stage);
+
+    ExecutionStatus taskStatus;
+    String message;
+
+    if (monitorDefinition.isFailOnError()) {
+      message =
+          "Deployment monitor failed to evaluate health in the allotted time, assuming failure because the monitor is configured to failOnError";
+      taskStatus = ExecutionStatus.TERMINAL;
+    } else {
+      message =
+          "Deployment monitor failed to evaluate health in the allotted time, proceeding anyway because the monitor is not configured to failOnError";
+      taskStatus = ExecutionStatus.FAILED_CONTINUE;
+    }
+
+    return buildTaskResult(TaskResult.builder(taskStatus), message);
   }
 
   @Override
   public @Nonnull TaskResult execute(@Nonnull Stage stage) {
     MonitoredDeployStageData context = stage.mapTo(MonitoredDeployStageData.class);
 
     try {
-      this.stage = stage;
-      this.monitorDefinition =
-          deploymentMonitorServiceProvider.getDefinitionById(
-              context.getDeploymentMonitor().getId());
-
+      ensureMonitorDefinition(stage);
       return executeInternal();
     } catch (RetrofitError e) {
+      log.error(
```

Comment:
wonder about logging this at `error` if this might be a something we retry on

Reply:
hm.. that's fair, let me change to `warn`